experimental/puppeteer-firefox/lib/DOMWorld.js

```diff
@@ -134,7 +134,7 @@ class DOMWorld {
   async setContent(html) {
     await this.evaluate(base64html => {
       document.open();
-      document.write(atob(base64html));
+      document.write(decodeURIComponent(atob(base64html).split('').map(c => '%' + ('00' + c.charCodeAt(0).toString(16)).slice(-2)).join('')));
```

Comment:
@semoal well, this is impressive.

I have hard time understanding what this does. Can you please explain this to me?

Reply:
Hi! basically, DOMString are 16-bit encoded strings, so when you call atob/btoa this methods create a exception when they exceeds the range of 8-bit byte (0x00~0xFF).
Mozilla MDN docs spoke a lot about this situation: https://developer.mozilla.org/en-US/docs/Web/API/WindowBase64/Base64_encoding_and_decoding

Normally ASCII characters like ™ if this is not handled, they got converted to something extrange like: â¢, because exceed the range of bytes allowed.

Reply:
> Hi! basically, DOMString are 16-bit encoded strings, so when you call atob/btoa this methods create a exception when they exceeds the range of 8-bit byte (0x00~0xFF).

@semoal I don't think `atob` actually throws an exception (at least it does not for me). I also don't think your solution corresponds to any of the solutions, listed on the [MDN page](https://developer.mozilla.org/en-US/docs/Web/API/WindowBase64/Base64_encoding_and_decoding#The_Unicode_Problem). 

So how do you actually fix this - from the high-level standpoint? Why don't you change the base64 encoding part?

Reply:
Ah ok, I think I understand now.

Can't you drop the `'00'` together with `slice(-2)`?

```js
document.write(decodeURIComponent(atob(base64html).split('').map(c => '%' + c.charCodeAt(0).toString(16)).join('')));
```

Reply:
Yes your solution works too mate!

Reply:
I've did a playground for testing online mate, https://repl.it/@semoal/InternationalGullibleDataset

You can see here all the solutions we have provided

Reply:
@semoal the `'00'` are actually important for things like a "newline" - I added a test for this.